### DISCUSSION TITLE
Debagger and debaggee JVM's  versions differ

### DISCUSSION THREAD

**Discussion by sbushmanov** [scalameta/metals#5191] [Q&A]
When trying to start a debug session in metals/dap in neovim, I'm getting the following message:

```
[Warn] The debugger and the debuggee are running in different versions of JVMs. You could see wrong source mapping results.
Debugger JVM version: 11.0.16
Debuggee JVM version: 1.8.0_302
```

My system wide Java version is 1.8.0 (i.e. JAVA_HOME)

I suspect some time ago I switched from Java 11 to Java 8, but now how/where should I set debugger JVM version?

**Comment by tgodzik** [comment]
Thanks for the question! If you are using Bloop then the Java version is set in `~/.bloop/bloop.json` and Metals should keep it synced with the Metals JVM version. You might need to restart the build server to get a correct version of JVM.

In most cases however this should not be an issue since debugee and debugger run in different processes. The only issue might be if you try to step into JDK methods.
